### PR TITLE
Composer improvements

### PR DESCRIPTION
**Pull request by Soullivaneuh**
* Use new composer constraints notation
* Add missing PHP constraint
* New sonata project majors requiring

### CODE REVIEW

**Comment by Soullivaneuh** [comment]
@sonata-project/contributors Can anyone explains me this issue? https://travis-ci.org/sonata-project/SonataAdminBundle/jobs/127012710#L554

File: composer.json [hunk 1]
```diff
@@ -17,36 +17,37 @@
...
+        "doctrine/inflector": "^1.0",
         "knplabs/knp-menu-bundle": "^2.1.1"
     },
     "require-dev": {
-        "jms/di-extra-bundle": "~1.7",
-        "sensio/generator-bundle": "~2.3|~3.0",
-        "symfony/yaml": "~2.3|~3.0",
+        "jms/di-extra-bundle": "^1.7",
+        "sensio/generator-bundle": "^2.3 || ^3.0",
+        "symfony/yaml": "^2.3 || ^3.0",
```
Comment (greg0ire) [inline composer.json]: Setting the `sort-package` option in all composer.json of all sonata-projects would be great.

Comment (Soullivaneuh) [inline composer.json]: What `sort-package` option does exactly?

Comment (greg0ire) [inline composer.json]: It sorts all package, but this works with `composer require` only. https://getcomposer.org/doc/06-config.md#sort-packages

Comment (Soullivaneuh) [inline composer.json]: Does it works with `composer validate` too?

Can be discussed on a sonata-project/dev-kit issue.

Comment (greg0ire) [inline composer.json]: > Does it works with composer validate too?

I tested that, but no, `validate` does not report anything about that.

**Pull request merged**